### ISSUE TITLE
[bug] misbehavior on lab.web3modal.com

### ISSUE THREAD

**Issue by mahnunchik** [reown-com/appkit#2559]
### Link to minimal reproducible example

https://lab.web3modal.com/

### Summary

Hi,

I'm trying to pass the tests listed in ["How do we test wallets?"](https://docs.walletconnect.com/cloud/explorer-submission#how-do-we-test-wallets), but I encountered incorrect behavior of `lab.web3modal.com` itself, which does not allow me to pass the tests to submit the wallet.

## Ethers

* Wallet https://coin.space/wallet/settings/walletconnect
* Dapp https://lab.web3modal.com/library/ethers/

Switching chains on dapp side doesn't reflect interface on dapp side.

![correct](https://github.com/user-attachments/assets/919aacb5-599b-48dd-bcc3-4981b8e99bb9)

After switching chain to Polygon.

![incorrect](https://github.com/user-attachments/assets/546483c0-d6fc-4155-b576-742e614cc01d)

## Wagmi

* Wallet https://coin.space/wallet/settings/walletconnect
* Dapp https://lab.web3modal.com/library/ethers-all/

Switching chains on dapp side doesn't reflect interface and logic on dapp side. So signing request comes to the wallet with incorrect data and is not completed successfully.

![w](https://github.com/user-attachments/assets/3291da81-85dc-4d53-9958-a5c567df31c9)

All tests work successfully with the first chain (Ethereum chain). But after switching chain everything breaks.

Regardless of which chain is selected before connecting, Ethereum is selected after a successful connection.

### List of related npm package versions

@apps/laboratory@5.0.7

**Comment by glitch-txs** [comment]
Why should the address change when switching chains? How does the account start you're sending to the dapp looks like?

**Comment by mahnunchik** [comment]
@rtomas Why it is closed? Issue doesn't solved.

@glitch-txs because it changes. Address may change or may not.

The wallet sends an array of accounts according to the specification, but the test app does not handle it.

```json
{
  "chains": [
    "eip155:421614",
    "eip155:43113",
    "eip155:97",
    "eip155:11155111",
    "eip155:63",
    "eip155:4002",
    "eip155:11155420",
    "eip155:80001"
  ],
  "accounts": [
    "eip155:421614:0xAd8eDE8697aF92AF31Fe66EDAD9A52A74F74464E",
    "eip155:43113:0xeb9248D23d31E1848B0331dBb4F527E6b8547768",
    "eip155:97:0x9Ab668f0818E371862886A9c6Dd1C3E07F31E583",
    "eip155:11155111:0x458B8967776485CCFEE42d8AD339752b53D9BC7B",
    "eip155:63:0xc6b07471826D3bA19B0D3A31E0a30b289F0d178c",
    "eip155:4002:0x4022F64e13ac127FDb85Fa8d6c6d725cb255032D",
    "eip155:11155420:0x0CFfE8EF251B2C8Ee2481F0142b531715877D417",
    "eip155:80001:0x9492A87Fc4c112Fe10123B1D3Aa6fBa5e14F1479"
  ]
}
```

**Comment by glitch-txs** [comment]
@mahnunchik if the address changes the wallet needs to emit the "accountsChanged" event, following EIP-1193

**Comment by glitch-txs** [comment]
Is this a Smart Account wallet?

**Comment by mahnunchik** [comment]
No, it is Dumb Account wallet which uses different addresses for each chain https://github.com/CoinSpace/CoinSpace/blob/master/web/src/lib/account/WalletConnect.js#L65-L92

**Comment by mahnunchik** [comment]
@rtomas Could you please reopen this issue.
